nistats/contrasts.py

```diff
@@ -312,6 +312,8 @@ def fixed_effects_img(contrast_imgs, variance_imgs, mask,
     # instantiate a masker
     if isinstance(mask, NiftiMasker):
         masker = mask
+    elif mask is None:
+        masker = NiftiMasker().fit(contrast_imgs)
```

Comment:
Would it be possible to add `smoothing_fwhm` as an optional kwarg? Some users may want to smoothing prior to this step.

It would also be similar to `SecondLevelModel` and `non_parametric_inference`

Maybe can copy this logic to apply it to both learned and passed in masks:
https://github.com/nistats/nistats/blob/f7e1056e26bd5b1070fc361a1d4335b14f0de32c/nistats/second_level_model.py#L635

Reply:
Also, a question. is doing `masker.fit(contrast_imgs)` equivalent to computing a mean img and fitting the mask to that (as is done in the SecondLevelModel)

Reply:
e.g.: 
```
    if isinstance(mask, NiftiMasker):
        masker = mask
    elif mask is None:
        masker = NiftiMasker(smoothing_fwhm=smoothing_fwhm).fit(contrast_imgs)
    else:
        masker = NiftiMasker(mask_img=mask, smoothing_fwhm=smoothing_fwhm).fit()
```

Minor comment: it seems other similar functions (such as `SecondLevelModel`, will call `.fit` even when a masker is passed, so this could confuse people)

Reply:
> Would it be possible to add `smoothing_fwhm` as an optional kwarg? Some users may want to smoothing prior to this step.

The problem is: what do you do with the variance ? you smooth it by the same amount ?
This is probably suboptimal (the variance obtained from the smoothed data would be smaller than the smoothed variance) and it makes it unclear what the statistic tests. 
So I'd rather heavily discourage it.

Reply:
Okay, I see. I suppose users can always pass a mask with their settings of choice if they want to do something else.